packages/core/strapi/src/services/document-service/index.ts

```diff
@@ -0,0 +1,281 @@
+    }
+
+    const entriesToDelete = await db.query(uid).findMany({
+      ...query,
+      where: {
+        ...params.lookup,
+        ...query?.where,
+        documentId,
+      },
+    });
+
+    // Delete all matched entries and its components
+    await mapAsync(entriesToDelete, async (entryToDelete: any) => {
+      const componentsToDelete = await getComponents(uid, entryToDelete);
+      await db.query(uid).delete({ where: { id: entryToDelete.id } });
+      await deleteComponents(uid, componentsToDelete as any, { loadComponents: false });
+    });
+
+    // TODO: Change return value to actual count
+    return entriesToDelete.at(0);
+  },
+
+  // TODO: should we provide two separate methods?
+  async deleteMany(uid, paramsOrIds) {
+    let queryParams;
+    if (isArray(paramsOrIds)) {
+      queryParams = { filter: { documentID: { $in: paramsOrIds } } };
+    } else {
+      queryParams = paramsOrIds;
+    }
+
+    const query = transformParamsToQuery(uid, queryParams || ({} as any));
+
+    return db.query(uid).deleteMany(query);
+  },
+
+  async create(uid, params) {
+    // TODO: Entity validator.
+    // TODO: File upload - Probably in the lifecycles?
+    const { data } = params;
+
+    if (params.status === 'published') {
+      throw new Error(
+        'Cannot directly create a published document. Use the publish method instead.'
+      );
+    }
+
+    if (!data) {
+      throw new Error('Create requires data attribute');
+    }
+
+    const model = strapi.getModel(uid) as Shared.ContentTypes[Common.UID.ContentType];
+
+    const componentData = await createComponents(uid, data);
+    const entryData = createPipeline(Object.assign(omitComponentData(model, data), componentData), {
+      contentType: model,
+    });
+
+    // select / populate
+    const query = transformParamsToQuery(uid, pickSelectionParams(params));
+
+    return db.query(uid).create({ ...query, data: entryData });
+  },
+
+  // NOTE: What happens if user doesn't provide specific publications state and locale to update?
+  async update(uid, documentId, params) {
+    // TODO: Prevent updating a published document
+    // TODO: Entity validator.
+    // TODO: File upload
+    const { data } = params || {};
+    const model = strapi.getModel(uid) as Shared.ContentTypes[Common.UID.ContentType];
+
+    if (params?.status === 'published') {
+      throw new Error('Cannot update a published document. Use the publish method instead.');
+    }
+
+    const query = transformParamsToQuery(uid, pickSelectionParams(params || {}));
+
+    // Find all locales of the document
+    const entries = await db
+      .query(uid)
+      .findMany({ ...query, where: { ...params?.lookup, ...query?.where, documentId } });
+
+    // Document does not exist
+    if (!entries.length) {
+      return null;
+    }
+
+    // TODO: How do we do this from i18n package?
+    const entryToUpdate = entries.find((entry) => entry.locale === params?.locale);
```

Comment:
I wonder how we could handle this in a clean way. I don't believe the logic to know if an entry has to be updated or created in a document should be in i18n.